Fix return value, invalid pointer to integer conversion with clang 15.

Index: src/libmpd-playlist.c
--- src/libmpd-playlist.c.orig
+++ src/libmpd-playlist.c
@@ -759,7 +759,7 @@ MpdData * mpd_playlist_search_commit(MpdObj *mi)
 	if(mpd_lock_conn(mi))
 	{
 		debug_printf(DEBUG_ERROR,"lock failed\n");
-		return NULL;
+		return MPD_LOCK_FAILED;
 	}
 	mpd_commitSearch(mi->connection);
 	while (( ent = mpd_getNextInfoEntity(mi->connection)) != NULL)
